autoPyTorch/pipeline/components/setup/network/base_network.py

```diff
@@ -105,22 +107,35 @@ def predict(self, loader: torch.utils.data.DataLoader) -> torch.Tensor:
         """
         Performs batched prediction given a loader object
         """
-        assert self.network is not None
-        self.network.eval()
-
+        if len(self.network_snapshots) == 0:
+            assert self.network is not None
+            return self._predict(network=self.network.float(), loader=loader).cpu().numpy()
```

Comment:
Why are you casting the network to float?

Reply:
as I converted the network to double earlier, for predict I am changing It back to float